src/parser.js

```diff
@@ -617,38 +617,95 @@
    * Used for caching SVG documents (loaded via `fabric.Canvas#loadSVGFromURL`)
    * @namespace
    */
-  var svgCache = {
+  var svgCache = (function() {
+    var cache = {},
+        requesting = {};
+
+    return {
+
+      /**
+       * @param {String} name
+       * @param {Function} callback
+       */
+      has: function (name, callback) {
+        callback(name in cache || name in requesting);
+      },
 
-    /**
-     * @param {String} name
-     * @param {Function} callback
-     */
-    has: function (name, callback) {
-      callback(false);
-    },
+      /**
+       * @param {String} name
+       * @param {Function} callback
+       */
+      get: function (name, callback) {
+        if (name in cache) {
+          callback(cache[name]);
+        }
+        else {
+          requesting[name].push(callback);
+        }
+      },
 
-    get: function () {
-      /* NOOP */
-    },
+      /**
+       * @param {String} name
+       * @param {Object} objToCache
+       */
+      set: function (name, objToCache) {
+        cache[name] = objToCache;
 
-    set: function () {
-      /* NOOP */
-    }
-  };
+        if (name in requesting && requesting[name].length > 0) {
+          requesting[name].forEach(function(cb) {
+            cb.call(this, objToCache);
+          });
+        }
+      },
+
+      /**
+       * @param {String} name
+       */
+      setRequesting: function(name) {
+        if (!(name in requesting)) {
+          requesting[name] = [];
+        }
+      }
+
+    };
+  })();
 
   /**
    * @private
    */
-  function _enlivenCachedObject(cachedObject) {
+  function _enlivenCachedObject(cachedObject, callback) {
 
     var objects = cachedObject.objects,
-        options = cachedObject.options;
-
-    objects = objects.map(function (o) {
-      return fabric[capitalize(o.type)].fromObject(o);
+        options = cachedObject.options,
+        objectsLength = objects.length,
+        builtObjects = {};
+
+    objects.forEach(function(o, i) {
+      var eachObjCallback = function(builtObj) {
+            addBuiltObject(builtObj, i);
+          },
+          result = fabric[capitalize(o.type)].fromObject(o, eachObjCallback);
+
+      if (result instanceof fabric.constructor) {
+        eachObjCallback(result);
+      }
     });
 
-    return ({ objects: objects, options: options });
+    function addBuiltObject(builtObj, objectIndex) {
+      builtObjects[objectIndex] = builtObj;
+
+      var builtObjectsLength = Object.keys(builtObjects).length;
```

Comment:
Also, I couldn't think of a better way to maintain the order of the objects, so I'm using `Object.keys` to tell when all of them are done. If we decided to keep it this way we may need to add an `Object.keys` polyfill into `src/util/lang_object.js`.